src/custom/state/lists/hooks/hooksMod.ts

```diff
@@ -124,35 +127,41 @@ export function useAllLists(): AppState['lists'][ChainId]['byUrl'] {
     }
     return memo
   }, {}) as TokenAddressMap
-} */
+}
+
+export const combineMaps = memoizeOne(_combineMaps)
 
 // merge tokens contained within lists from urls
 export function useCombinedTokenMapFromUrls(urls: string[] | undefined): TokenAddressMap {
   // MOD: recreated here to allow it to use the custom useAllLists hook
   const lists = useAllLists()
-  return useMemo(() => {
-    if (!urls) return EMPTY_LIST
-
-    return (
-      urls
-        .slice()
-        // sort by priority so top priority goes last
-        .sort(sortByListPriority)
-        .reduce((allTokens, currentUrl) => {
-          const current = lists?.[currentUrl]?.current
-          if (!current) return allTokens
-          try {
-            const newTokens = Object.assign(listToTokenMap(current))
-            return combineMaps(allTokens, newTokens)
-          } catch (error) {
-            console.error('Could not show token list due to error', error)
-            return allTokens
-          }
-        }, EMPTY_LIST)
-    )
-  }, [lists, urls])
+
+  return useMemo(() => memoizedCombinedTokenMapFromUrls(urls, lists), [urls, lists])
+}
+
+function combinedTokenMapFromUrls(urls: string[] | undefined, lists: any): TokenAddressMap {
+  if (!urls) return EMPTY_LIST
+  return (
+    urls
+      .slice()
+      // sort by priority so top priority goes last
+      .sort(sortByListPriority)
+      .reduce((allTokens, currentUrl) => {
+        const current = lists?.[currentUrl]?.current
+        if (!current) return allTokens
+        try {
+          const newTokens = Object.assign(listToTokenMap(current))
+          return combineMaps(allTokens, newTokens)
```

Comment:
If I understood correctly:
- combineMaps will memoize only the last result based on the passed arguments
- combineMaps is executed within a loop/reducer
- thus, it'll only have a cache hit if the accumulator (`allTokens`) and the tokens from the new list (`newTokens`) are identical.

Is that it?
Is it likely to get a cache hit at all or did I get this wrong?

Reply:
Yes, you are right. I changed the implementation so the combineMaps in this case is called just once with all the tokens maps and merge them in one call.